examples/test/test_qnn_tooklit.py

```diff
@@ -6,64 +6,88 @@
+        target_path = tmp_path
+        blob, download_path = "", ""
+        if platform.system() == "Windows":
+            blob, download_path = "qnn_sdk_windows.zip", "qnn_sdk_windows.zip"
+        elif platform.system() == "Linux":
+            blob, download_path = "qnn_sdk_linux.zip", "qnn_sdk_linux.zip"
 
-    download_azure_blob(
-        container="olivetest",
-        blob=blob,
-        download_path=download_path,
-    )
-    if not target_path:
+        download_azure_blob(
+            container="olivetest",
+            blob=blob,
+            download_path=download_path,
+        )
         target_path = Path().resolve() / "qnn_sdk"
-    target_path.mkdir(parents=True, exist_ok=True)
-    if platform.system() == "Windows":
-        cmd = f"powershell Expand-Archive -Path {download_path} -DestinationPath {str(target_path)}"
-        run_subprocess(cmd=cmd, check=True)
-    elif platform.system() == "Linux":
-        run_subprocess(cmd=f"unzip {download_path} -d {str(target_path)}", check=True)
-    return target_path
+        target_path.mkdir(parents=True, exist_ok=True)
+        if platform.system() == "Windows":
+            cmd = f"powershell Expand-Archive -Path {download_path} -DestinationPath {str(target_path)}"
+            run_subprocess(cmd=cmd, check=True)
+        elif platform.system() == "Linux":
+            run_subprocess(cmd=f"unzip {download_path} -d {str(target_path)}", check=True)
 
+        os.environ["QNN_SDK_ROOT"] = str(target_path / "opt" / "qcom" / "aistack")
 
-def setup_resource():
-    """Setups any state specific to the execution of the given module."""
-    cur_dir = Path(__file__).resolve().parent.parent
-    example_dir = cur_dir / "mobilenet"
-    os.chdir(example_dir)
+    def _setup_resource(self, use_olive_env):
+        """Setups any state specific to the execution of the given module."""
+        cur_dir = Path(__file__).resolve().parent.parent
+        example_dir = cur_dir / "mobilenet"
+        os.chdir(example_dir)
 
-    # prepare model and data
-    # retry since it fails randomly
-    run_subprocess(cmd="python -m olive.platform_sdk.qualcomm.configure --py_version 3.8 --sdk qnn", check=True)
-    # install dependencies
-    python_cmd = ""
-    if platform.system() == "Windows":
-        python_cmd = str(Path(os.environ["QNN_SDK_ROOT"]) / "olive-pyenv" / "python.exe")
-    elif platform.system() == "Linux":
-        python_cmd = str(Path(os.environ["QNN_SDK_ROOT"]) / "olive-pyenv" / "bin" / "python")
-    install_cmd = [
-        python_cmd,
-        str(Path(os.environ["QNN_SDK_ROOT"]) / "bin" / "check-python-dependency"),
-    ]
-    run_subprocess(cmd=" ".join(install_cmd), check=True)
-    retry_func(run_subprocess, kwargs={"cmd": "python download_files.py", "check": True})
-    retry_func(run_subprocess, kwargs={"cmd": "python prepare_config.py --use_raw_qnn_sdk", "check": True})
+        if use_olive_env:
+            os.environ["USE_OLIVE_ENV"] = "1"
```

Comment:
not part of this PR but it might be nice to provide an option in the local system config to set environment variables like https://github.com/microsoft/Olive/blob/25eec4950f34e7db4d8f75a4b1ba8d6c858c80e5/olive/systems/python_environment/python_environment_system.py#L37 
This way, during the local system initialization, we can update `os.environ`. Also lets the user know they set the variable correctly.

Reply:
Yes, good suggestion. 👍
I think that before actually based on the purpose like: how to let user directly set the SNPE_ROOT or other required options in Olive config? User always expects the simplest way without any other efforts. Just config~
Let me add the TODO item.